starfish/image/_stack.py

```diff
@@ -45,6 +52,23 @@ def numpy_array(self, data):
         self._data_needs_writeback = True
         data.setflags(write=False)
 
+    def slice(self, indices):
+        slice_list = [slice(None, None, None) for _ in range(ImageStack.MAX_INDEX)]
+        for name, val in indices.items():
+            idx = ImageStack.INDEX_MAP[name]
+            slice_list[idx] = val
+        result = self._data[tuple(slice_list)]
+        result.setflags(write=False)
+        return result
+
+    def set_slice(self, indices, data):
+        slice_list = [slice(None, None, None) for _ in range(ImageStack.MAX_INDEX)]
+        for name, val in indices.items():
+            idx = ImageStack.INDEX_MAP[name]
+            slice_list[idx] = val
+        self._data[tuple(slice_list)] = data
+        self._data_needs_writeback = True
+
     @property
     def shape(self):
```

Comment:
Given our switch over to named dimensions, can we have shape output explicit dimension names instead of just (8, 2, 34, 2048, 2048)

e.g. 
```
> self.shape
HYB: 8
CH: 2
Z: 34
X: 2048
Y: 2048
```

Reply:
i like that.

Reply:
#137